--- a/blackmagic-desktop-video-10.9/usr/src/blackmagic-io-10.9a7/bm_mm.c	2017-04-04 00:54:02.000000000 -0400
+++ b/blackmagic-desktop-video-10.9/usr/src/blackmagic-io-10.9a7/bm_mm.c	2017-04-22 14:37:57.395802405 -0400
@@ -103,7 +103,12 @@
 		return false;

 	down_read(&task->mm->mmap_sem);
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 10, 0)
+		if (task == current)
+			ret = get_user_pages((unsigned long)address & PAGE_MASK, umem->length, write ? FOLL_WRITE : 0, umem->pages, NULL);
+		else
+			ret = get_user_pages_remote(task, task->mm, (unsigned long)address & PAGE_MASK, umem->length, write ? FOLL_WRITE : 0, umem->pages, NULL, NULL);
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 0)
 		if (task == current)
 			ret = get_user_pages((unsigned long)address & PAGE_MASK, umem->length, write ? FOLL_WRITE : 0, umem->pages, NULL);
 		else

